backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ExportedLibraries.kt

```diff
@@ -7,25 +7,78 @@ package org.jetbrains.kotlin.backend.konan
 
 import org.jetbrains.kotlin.cli.common.messages.CompilerMessageSeverity
 import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.descriptors.konan.CurrentKonanModuleOrigin
+import org.jetbrains.kotlin.descriptors.konan.DeserializedKonanModuleOrigin
+import org.jetbrains.kotlin.descriptors.konan.SyntheticModulesOrigin
+import org.jetbrains.kotlin.descriptors.konan.konanModuleOrigin
 import org.jetbrains.kotlin.konan.file.File
+import org.jetbrains.kotlin.konan.library.KonanLibrary
+import org.jetbrains.kotlin.konan.library.isInterop
+import org.jetbrains.kotlin.konan.library.resolver.KonanLibraryResolveResult
 
-internal fun validateExportedLibraries(configuration: CompilerConfiguration, libraryFiles: List<File>) {
-    val exportedLibraries = getExportedLibraries(configuration)
+internal fun validateExportedLibraries(configuration: CompilerConfiguration, resolvedLibraries: KonanLibraryResolveResult) {
+    getExportedLibraries(configuration, resolvedLibraries, report = true)
+}
+
+internal fun Context.getExportedDependencies(): List<ModuleDescriptor> {
+    val exportedLibraries = getExportedLibraries(this.config.configuration, this.config.resolvedLibraries, report = false)
+            .toSet()
+
+    val result = this.moduleDescriptor.allDependencyModules.filter {
+        val origin = it.konanModuleOrigin
+        when (origin) {
+            CurrentKonanModuleOrigin, SyntheticModulesOrigin -> false
+            is DeserializedKonanModuleOrigin -> {
+                origin.library in exportedLibraries
+            }
+        }
+    }
+
+    return result
+}
 
-    val missingExportedLibraries = exportedLibraries - libraryFiles
+private fun getExportedLibraries(
+        configuration: CompilerConfiguration,
+        resolvedLibraries: KonanLibraryResolveResult,
+        report: Boolean
+): List<KonanLibrary> {
+    val exportedLibraries = configuration.getList(KonanConfigKeys.EXPORTED_LIBRARIES).map { File(it) }.toSet()
+    val remainingExportedLibraries = exportedLibraries.toMutableSet()
 
-    if (missingExportedLibraries.isNotEmpty()) {
+    val result = mutableListOf<KonanLibrary>()
+
+    val libraries = resolvedLibraries.getFullList(null)
+
+    for (library in libraries) {
+        val libraryFile = library.libraryFile
+        if (libraryFile in exportedLibraries) {
+            remainingExportedLibraries -= libraryFile
+            if (library.isInterop || library.isDefault) {
+                if (report) {
+                    val kind = if (library.isInterop) "Interop" else "Default"
+                    configuration.report(
```

Comment:
Not sure if this limitation makes sense for interop libs. I guess it shall be limited to stdlib.